applications/conceptual-model-editor/src/app/diagram/reactflow/components/menu-options.tsx

```diff
@@ -43,40 +45,74 @@ export const useMenuOptions = () => {
         position?: string;
     }) => {
-        const { openDetailHandler, createProfileHandler, modifyHandler, removeFromViewHandler, deleteHandler } = props;
+        const { openDetailHandler, createProfileHandler, modifyHandler, removeFromViewHandler, deleteHandler, position } = props;
+        // You have to actually define the type here explictly, typescript can't interfere the type correctly
+        const handlersWithTexts: Array<[(() => void) | undefined, string]> = [      
+            [openDetailHandler, "ℹ Detail"],
+            [modifyHandler, "✏ Modify"],
+            [undefined, "HorizontalSeparator"],
+            [createProfileHandler, "🧲 Create profile"],            
+            [undefined, "HorizontalSeparator"],
+            [removeFromViewHandler, "🕶 Remove from view"],
+            [deleteHandler, "🗑 Delete"],
+        ];
+        return MenuOptionsGeneral({handlersWithTexts, positionTailwind: position});        
+    };    
+
+    /**
+     * 
+     * @param props is an object with following properties:
+     * - handlersWithTexts is an array of tuples. The first property specifies the handler called on click and the second one the name. 
+     * Special case is if the handler is undefined and the name is "HorizontalSeparator".
+     * - positionTailwind specifies the position in tailwind notation.
+     * - positionCSS specifies the position in CSS notation. If both the tailwind and CSS positions are specified, the behavior is undefined.
+     * @returns
+     */
+    const MenuOptionsGeneral = (props: {
+        handlersWithTexts: Array<[(() => void) | undefined, string]>;
+        positionTailwind?: string;
+        positionCSS?: object;       // TODO: It might be possible to specify better type
+    }) => {        
+        const buttonsToRender = props.handlersWithTexts.map(e => {
+            const [handler, text] = e;
+            return ((text === "HorizontalSeparator" && handler === undefined) ? (<HorizontalSeparator />) :
+                (handler === undefined) ? (<></>) : (<MenuButton text={text} onClick={handler} />));                               
+        });              
+        
+        
+        let style: React.CSSProperties = { pointerEvents: "all" };
+        if (props.positionCSS !== undefined) {
+            style = { 
+                ...style, 
+                ...props.positionCSS
+            };
+        }
         return (
             <div
                 ref={menuOptionsRef}
                 tabIndex={-1}
-                style={{ pointerEvents: "all" }}
-                className={`flex flex-col bg-white border-2 border-slate-400 border-solid [&>*]:px-5 [&>*]:text-left ${props.position ? props.position : ""}`}
+                style={style}
+                className={`flex flex-col bg-white border-2 border-slate-400 border-solid [&>*]:px-5 [&>*]:text-left ${props.positionTailwind ? props.positionTailwind : ""}`}                
                 onBlur={(event) => {
```

Comment:
Again .. I would like us to have a convention that we define all handlers, function, callback as variables outside the JSX. There might be exceptions for map ... What do you think?

Reply:
Not sure if I understand. Do you mean like for example instead of just having visualization.tsx we would have visualization.tsx where we would have only the rendering and maybe some state variables and everything else would come from other files (for example: visualization-alignment.ts, visualization-helper-methods.ts, visualization-callbacks.ts, ...).